lib/kms/wrapper.js

```diff
@@ -0,0 +1,246 @@
+                    log.warn('error from kms', { implName, error: err });
+                    return cb(err);
+                }
+                log.trace('decipher created by the kms');
+                return cb(null, decipher);
+            });
+    }
+
+    /*
+     * cb(err, cipherBundle)
+     */
+    static createCipherBundle(serverSideEncryption,
+                              log, cb) {
+        const dataKey = this.createDataKey(log);
+        const cipherBundle = {
+            algorithm: serverSideEncryption.algorithm,
+            masterKeyId: serverSideEncryption.masterKeyId,
+            cryptoScheme: 1,
+            cipheredDataKey: null,
+            cipher: null,
+        };
+
+        async.waterfall([
+            function cipherDataKey(next) {
+                return kms._cipherDataKey(cipherBundle.cryptoScheme,
+                                          serverSideEncryption.masterKeyId,
+                                          dataKey, log, next);
+            },
+            function createCipher(cipheredDataKey, next) {
+                cipherBundle.cipheredDataKey =
+                    cipheredDataKey.toString('base64');
+                return kms._createCipher(cipherBundle.cryptoScheme,
+                                         dataKey, 0, log, next);
+            },
+            function finishCipherBundle(cipher, next) {
+                cipherBundle.cipher = cipher;
+                return next(null, cipherBundle);
+            },
+        ], (err, cipherBundle) => {
+            if (err) {
+                log.error('error processing cipher bundle',
+                          { implName, error: err });
+            }
+            return cb(err, cipherBundle);
+        });
+    }
+    /*
+     * cb(err, decipherBundle)
+     */
+    static createDecipherBundle(serverSideEncryption, offset,
+                                log, cb) {
+        const decipherBundle = {
+            cryptoScheme: serverSideEncryption.cryptoScheme,
+            decipher: null,
+        };
+        if (!serverSideEncryption.masterKeyId ||
+            !serverSideEncryption.cipheredDataKey ||
+            !serverSideEncryption.cryptoScheme) {
+            log.error('Invalid cryptographic information', { implName });
+            return cb(errors.InternalError);
+        }
+        async.waterfall([
+            function decipherDataKey(next) {
+                return kms._decipherDataKey(
+                    decipherBundle.cryptoScheme,
+                    serverSideEncryption.masterKeyId,
+                    serverSideEncryption.cipheredDataKey,
+                    log, next);
+            },
+            function createDecipher(plainTextDataKey, next) {
+                return kms._createDecipher(decipherBundle.cryptoScheme,
+                                           plainTextDataKey,
+                                           offset, log, next);
+            },
+            function finishDecipherBundle(decipher, next) {
```

Comment:
You could combine this with the final callback (or leave as is if you think clearer).